### PR TITLE
Enable HPA for Airflow Webserver

### LINKED ISSUE #41876: Add Horizontal Pod Autoscaler (HPA) to webserver deployment
### Description

Add a Horizontal Pod Autoscaler (HPA) to the webserver deployment. Currently, we have an HPA in the worker deployment with a conditional statement that disables replicas in that deployment, because attempting to use HPA without disabling replicas could cause a conflict. I want to add an HPA to the webserver deployment to scale it up during high loads.

### Use case/motivation

If someone has a lot of users who use the UI, they will want to use HPA instead of replicas to reduce costs.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by JoaVirtudes19**
closes [#41876](https://github.com/apache/airflow/issues/41876)
Add a Horizontal Pod Autoscaler (HPA) to the webserver for hight loads of users.

### CODE REVIEW

**Comment by JoaVirtudes19** [comment]
@jedcunningham

File: helm_tests/webserver/test_webserver.py [hunk 1]
```diff
@@ -1154,3 +1154,86 @@ def test_overridden_automount_service_account_token(self):
             show_only=["templates/webserver/webserver-serviceaccount.yaml"],
         )
         assert jmespath.search("automountServiceAccountToken", docs[0]) is False
+
+
+class TestWebserverHPAAutoScaler:
+    """Tests webserver HPA auto scaler."""
+
+    def test_should_add_component_specific_labels(self):
```
Comment (jedcunningham) [inline helm_tests/webserver/test_webserver.py]: Can you move this to `test_hpa_webserver`?

File: helm_tests/webserver/test_webserver.py [hunk 2]
```diff
@@ -1154,3 +1154,86 @@ def test_overridden_automount_service_account_token(self):
...
+                    "type": "Pods",
+                    "pods": {
+                        "metric": {"name": "custom"},
+                        "target": {"type": "Utilization", "averageUtilization": 80},
+                    },
+                },
+            ),
+        ],
+    )
+    def test_should_use_hpa_metrics(self, metrics, expected_metrics):
```
Comment (jedcunningham) [inline helm_tests/webserver/test_webserver.py]: Same with this one?

File: helm_tests/webserver/test_webserver.py [hunk 3]
```diff
@@ -1154,3 +1154,86 @@ def test_overridden_automount_service_account_token(self):
...
+            values={
+                "webserver": {
+                    "hpa": {"enabled": True},
+                },
+            },
+            show_only=["templates/webserver/webserver-deployment.yaml"],
+        )
+        assert "replicas" not in jmespath.search("spec", docs[0])
+
+    def test_should_not_remove_replicas_field(self):
```
Comment (jedcunningham) [inline helm_tests/webserver/test_webserver.py]: This is probably better done by adding test coverage that `webserver.replicas` works. Can you remove this test and add coverage for that instead in `TestWebserverDeployment`?

File: helm_tests/webserver/test_webserver.py (hunk 1 above)
Comment (JoaVirtudes19) [reply helm_tests/webserver/test_webserver.py]: Sure!

File: helm_tests/webserver/test_webserver.py (hunk 2 above)
Comment (JoaVirtudes19) [reply helm_tests/webserver/test_webserver.py]: Done!

File: helm_tests/webserver/test_webserver.py (hunk 3 above)
Comment (JoaVirtudes19) [reply helm_tests/webserver/test_webserver.py]: I didn't realize, but seems to be a better place for these tests actually!

Comment (JoaVirtudes19) [reply helm_tests/webserver/test_webserver.py]: Done!

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks @JoaVirtudes19! Congrats on yours first PR 🎉

**Pull request merged**